examples/js/loaders/GLTFLoader.js

```diff
@@ -1298,6 +1301,60 @@ THREE.GLTFLoader = ( function () {
 
 	}
 
+	function isPrimitiveEqual ( a, b ) {
+
+		if ( a.indices !== b.indices ) {
+
+			return false;
+
+		}
+
+		var attribA = a.attributes || {};
+		var attribB = b.attributes || {};
+		var keysA = Object.keys(attribA);
+		var keysB = Object.keys(attribB);
+		var i;
+
+		for ( i = 0; i < keysA.length; i++ ) {
+
+			var key = keysA[i];
+
+			if ( attribA[key] !== attribB[key] ) {
+
+				return false;
+
+			}
+		}
+
+		for ( i = 0; i < keysB.length; i++ ) {
+
+			var key = keysB[i];
+
+			if ( attribA[key] !== attribB[key] ) {
+
+				return false;
+
+			}
+		}
+
```

Comment:
Probably, if you check `if ( keysA.length !== keysB.length ) return false;` you could remove `for ( i = 0; i < keysB.length; i++ ) { ... }`?

Reply:
I think both would be needed, in case one object has extra attributes that don't appear in the other. In that case they should be treated as two separate BufferGeometries.

Reply:
>  in case one object has extra attributes that don't appear in the other. 

I think `keysA.length` doesn't equal to `keysB.length` in that case.

Reply:
Ah yes , I misread your comment. 👍